lesson9/index.js

```diff
@@ -0,0 +1,46 @@
+const ul = document.getElementById('js-lists');
+
+const listData = [
+    {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+    {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+]
+
+function displayLoadingImage () {
+    const lodingImage = document.createElement('img');
+    lodingImage.id = 'loadingImageId';
+    lodingImage.src = './loading-circle.gif';
+    ul.appendChild(lodingImage);
+}
+
+async function loadingPromise() {
+    return new Promise(resolve => {
+        displayLoadingImage ();
+        setTimeout(() => {
+            resolve(listData);
+            const lodingImage = document.getElementById('loadingImageId');
+            lodingImage.remove();
+        }, 3000);
+    });
+}
+
+async function createLists() {
+    const frag = new DocumentFragment();
+    const value = await loadingPromise();
+    for(const list of value) {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        a.href = list.to;
+        const img = document.createElement('img');
+        img.src = list.img;
+        img.alt = list.alt;
+        const text = document.createTextNode(list.text);
+
+        a.appendChild(img);
+        a.appendChild(text);
+        li.appendChild(a);
+        frag.appendChild(li);
+    }
+    ul.appendChild(frag);
+}
+
+createLists();
```

Comment:
nits
The non-empty source file must end with a newline character, so a warning is issued. If you would like, you can refer to this page to fix it.
[reference article](https://qiita.com/facultyoflaw11/items/14c16f1b624ffd2f383c)

Reply:
Thank you for your reviewing!!
I didn't know about "No Newline at End of File".
And thank you for sharing the article.
I set it on my VScode!!

I end with a newline character, but I can't confirm on github......
Is it OK?

commit: 89155a6

Reply:
Isn't it perfect!
I'm glad I had you set it up because once it's set up, I don't have to think about it anymore!😊